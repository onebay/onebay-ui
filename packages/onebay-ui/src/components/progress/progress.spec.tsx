import { mount } from '@vue/test-utils'
import Progress from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Progress, {
    components: {},
    slots,
    props: { ...values },
  })
}

describe('Progress Snap', () => {
  it('render initial Progress', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props percent', () => {
    const wrapper = factory({ percent: 25 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props isHidePercent', () => {
    const wrapper = factory({ isHidePercent: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props color', () => {
    const wrapper = factory({ color: '#FF4949' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props status(progress)', () => {
    const wrapper = factory({ status: 'progress' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props status(error)', () => {
    const wrapper = factory({ status: 'error' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Progress -- props status(success)', () => {
    const wrapper = factory({ status: 'success' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
