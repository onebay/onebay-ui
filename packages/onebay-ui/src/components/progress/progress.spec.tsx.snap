// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Snap render Progress -- props color 1`] = `
<div
  class="ob-progress"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px; background-color: rgb(255, 73, 73);"
      />
    </div>
  </div>
  <!---->
</div>
`;

exports[`Progress Snap render Progress -- props isHidePercent 1`] = `
<div
  class="ob-progress"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </div>
  </div>
  <div
    class="ob-progress__content"
  >
    <div>
      0%
    </div>
  </div>
</div>
`;

exports[`Progress Snap render Progress -- props percent 1`] = `
<div
  class="ob-progress"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 25%; height: 10px;"
      />
    </div>
  </div>
  <!---->
</div>
`;

exports[`Progress Snap render Progress -- props status(error) 1`] = `
<div
  class="ob-progress ob-progress--error"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </div>
  </div>
  <!---->
</div>
`;

exports[`Progress Snap render Progress -- props status(progress) 1`] = `
<div
  class="ob-progress ob-progress--progress"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </div>
  </div>
  <!---->
</div>
`;

exports[`Progress Snap render Progress -- props status(success) 1`] = `
<div
  class="ob-progress ob-progress--success"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </div>
  </div>
  <!---->
</div>
`;

exports[`Progress Snap render initial Progress 1`] = `
<div
  class="ob-progress"
>
  <div
    class="ob-progress__outer"
  >
    <div
      class="ob-progress__outer-inner"
    >
      <div
        class="ob-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </div>
  </div>
  <!---->
</div>
`;
